Library/Homebrew/tap.rb

```diff
@@ -276,9 +276,10 @@ def remote
   def remote_repository
     return unless (remote = self.remote)
 
-    @remote_repository ||= remote.delete_prefix("https://github.com/")
-                                 .delete_prefix("git@github.com:")
+    @remote_repository ||= remote.delete_prefix("https://")
+                                 .delete_prefix("git@")
                                  .delete_suffix(".git")
+                                 .split(%r{[:\/]})[-2..].join("/")
```

Comment:
Can you clarify a bit more what this is doing? It's pretty hard to follow.

Reply:
Can you use one of the existing regexes e.g. `HOMEBREW_TAP_FORMULA_REGEX` or similar for this instead?

Reply:
None of the existing regexes in `tap_constants.rb` or anywhere else catch my eye. `HOMEBREW_TAP_FORMULA_REGEX` matches only a tap and formula in format `owner/repo/formula`. Here we need to extract the owner and repository name from strings like:
- `git@github.com:owner/repo`
- `https://github.com/owner/repo`

Also any custom remote should be supported (not breaking the PR I've linked above) like:
- `https://mirror.something.com/github/owner/repo.git`

Reply:
@dawidd6 Can you add a new regex in that format then with named captures? Much easier to follow.

Reply:
Done and seems to work as expected :slightly_smiling_face:.